speasy/core/codecs/bundled_codecs/hapi_csv/codec.py

```diff
@@ -0,0 +1,131 @@
+from typing import List, AnyStr, Optional, Mapping, Union, Dict, Any
+import io
+import logging
+
+from datetime import timedelta
+import numpy as np
+
+from speasy.core.codecs import CodecInterface, register_codec
+from speasy.core.codecs.codec_interface import Buffer
+from speasy.core.cache import CacheCall
+from speasy.products import SpeasyVariable, VariableTimeAxis, DataContainer
+from .csv_file import HapiCsvFile, HapiCsvParameter
+
+log = logging.getLogger(__name__)
+from .reader import load_hapi_csv
+from .writer import save_hapi_csv
+
+
+def _create_meta(variable:SpeasyVariable) -> Dict[str, Any]:
+    meta = {
+        "name": variable.name,
+        "units": variable.unit,
+        "fill": variable.fill_value,
+        "description": variable.meta.get("description", "")
+    }
+    if  np.issubdtype(variable.values.dtype, np.integer):
+        meta["type"] = "int"
+    elif np.issubdtype(variable.values.dtype, np.floating):
+        meta["type"] = "double"
+    else:
+        raise ValueError(f"Unsupported data type {variable.values.dtype}")
+
+    labels  =  variable.columns
+    if labels is not None and len(labels) > 0:
+        meta["label"] = labels
+
+    if "coordinateSystemName" in variable.meta:
+        meta["coordinateSystemName"] = variable.meta["coordinateSystemName"]
+    if "vectorComponents" in variable.meta:
+        meta["vectorComponents"] = variable.meta["vectorComponents"]
+
+    if  len(variable.values.shape) > 1:
+        meta["size"] = variable.values.shape[1:]
+    return meta
+
+
+def _decode_meta(meta: Dict[str, Any]) -> Dict[str, Any]:
+    if "units" in meta:
+        meta["UNITS"] = meta.pop("units")
+    return meta
+
+
+def _same_time_axis(variables: List[SpeasyVariable]) -> bool:
+    if len(variables) < 2:
+        return True
+    ref_time_axis = variables[0].time
+    return all([np.all(var.time == ref_time_axis) for var in variables[1:]])
+
+
+def _hapi_csv_to_speasy_variables(hapi_csv_file: HapiCsvFile, variables: List[AnyStr]) -> Mapping[str, SpeasyVariable]:
+    time_axis = VariableTimeAxis(values=hapi_csv_file.time_axis, meta=hapi_csv_file.time_axis_meta)
+    loaded_vars = {}
+    for var_name in variables:
+        parameter = hapi_csv_file.get_parameter(var_name)
+        if parameter is not None:
+            loaded_vars[var_name] = SpeasyVariable(axes=[time_axis], values=DataContainer(parameter.values,
```

Comment:
You only defined "time axis", but for a spectrogram you can have also a "VariableAxis".

For example: https://amda.irap.omp.eu/service/hapi/data?id=clust1-hia-pad&parameters=c1_hia_pad_omni&time.min=2020-06-16T06:00:00Z&time.max=2020-06-16T06:10:00.000Z&include=header